### DISCUSSION TITLE
got error and warning when running test codes for the legacy and new GUI visualizer

### DISCUSSION THREAD

**Discussion by hopezh** [isl-org/Open3D#6981] [Q&A]
As indicated [here](https://www.open3d.org/docs/release/arm.html), `open3d` support `ARM` architecture, e.g. Apple M1. 

However, I got warning running the test code for the `new GUI visualizer`:

```shell
# Test the new GUI visualizer
python -c "import open3d as o3d; c = o3d.geometry.TriangleMesh.create_box(); o3d.visualization.draw(c)"
```

The warning I have is:
<img width="1176" alt="Screenshot 2024-09-23 at 10 02 11 PM" src="https://github.com/user-attachments/assets/7d1ee486-7f71-4197-851c-1dbe4ac4cf98">

As to the test of the `legacy visualizer` I can get the GUI open, but there is a `open3d warning`, too:
<img width="1311" alt="Screenshot 2024-09-23 at 10 03 49 PM" src="https://github.com/user-attachments/assets/e600c1ff-9157-459d-9f57-315dcefac9f8">

May I ask if I have missing something important here? 

Thanks.

**Comment by rxba** [comment]
Hey @hopezh , the segfault may be related to your numpy version being >2.0, which is not yet supported by versions <=0.18.0 of Open3D. You can try downgrading numpy with `pip install numpy==1.26.4` and see if it fixes the issue.
The warning is a MacOS specific issue and can be ignored. It will be fixed with #6587 .
